packages/compass-aggregations/src/modules/import-pipeline.js

```diff
@@ -190,14 +189,17 @@ export const createNew = () => ({
 });
 
 /**
- * Confirm new action creator.
- *
- * @returns {Object} The state.
+ * @returns {import('.').PipelineBuilderThunkAction<void>}
  */
-export const confirmNew = () => (dispatch, getState) => {
-  const { importPipeline } = getState();
-  const pipeline = createPipeline(importPipeline.text);
-  const error = pipeline.length > 0 ? pipeline[0].syntaxError : null;
+export const confirmNew = () => (dispatch, getState, { pipelineBuilder }) => {
+  const { importPipeline: { text } } = getState();
```

Comment:
Using new pipeline parser here to parse and validate the text based pipeline that user pastes. 
(This will all eventually go away with the implementation of Pipeline As Text feature.)